public/_includes/_util-fns.jade

```diff
@@ -0,0 +1,35 @@
+- var getFrag = function(fileName) {
+-   var frag = partial(fileName);
+-   if (frag == null) {
+-     return "BAD FILENAME: " + fileName + "   Current path: " + current.path;
+-   } else {
+-     // ``` gets translated to <pre><code>.....</code></pre> and we need
+-     // to remove this from the fragment prefix is 11 long and suffix is 13 long
+-     var r = frag.substring(11, frag.length-13);
+-     return r;
+-   }
+- }
+
+- var getExtn = function(fileName) {
+-   var ix = fileName.lastIndexOf('.');
+-   return ix > 0 ? fileName.substr(ix+1) : "";
+- }
+
+// HACK: to avoid having to include a path in makeTabs calls
+- var currentPath = current.path;
+// need to back up to 'docs'
+- var pathToFrags = "../../../../../../../../../../../".substr(0, (currentPath.length-2)*3) + "_fragments/";
```

Comment:
I have trouble following this – you want to add `../` once for each of `currentPath`'s length minus 2? Would currentPath ever have a length > 13? Can you please add more explanation here, and maybe throw for cases when `currentPath` has a length that would be out of bounds?

Maybe an example of the expected values for the preceding couple of lines.